controller/src/lib.rs

```diff
@@ -124,73 +194,153 @@ fn schedule_response_for_existing_backend(
 }
 
 async fn process_response(
-    state: &StateHandle,
+    state: &mut StateHandle,
     sr: &ScheduleRequest,
     scheduler: &Scheduler,
     nats: &TypedNats,
 ) -> anyhow::Result<ScheduleResponse> {
     tracing::info!("checking locks");
     let cluster_name = sr.cluster.clone();
-    if let Some(lock_name) = sr.lock.clone() {
+    let schedule_response: Option<_> = if let Some(lock_name) = sr.lock.clone() {
         tracing::info!(?lock_name, "scheduling lock");
 
-        if let Ok(backend) = backend_of_lock(state, &cluster_name, &lock_name) {
-            tracing::info!(?backend, "fetch preexisting backend");
-            return schedule_response_for_existing_backend(state, cluster_name, backend);
-        }
+        if let Ok(Some(backend)) = backend_assigned_to_lock(state, &lock_name, &cluster_name) {
+            Some(schedule_response_for_existing_backend(
+                state,
+                cluster_name.clone(),
+                backend,
+            ))
+        } else {
+            let mut rng = rand::rngs::StdRng::from_entropy();
```

Comment:
not Send :/

do you know if there's a better solution that is?

Reply:
It shouldn't need to be Send because we don't need to hold it across an `await`, but it needs to be in its own block so that the compiler knows that, e.g. I think this should work:

```rust
let my_uid: u64 = {
    let rng = thread_rng();
    rng.gen()
};
```